### PR TITLE
Feature/#40 save update vehicles

### PR DESCRIPTION
**Pull request by camoore1979**
I believe I've completed the features intended to be implemented by @gwenf in her [branch](https://github.com/fusionalliance/autorenter-react/tree/feature/gwen-vehicles) and [PR](https://github.com/fusionalliance/autorenter-react/pull/44). I created a new branch for my work, and now this fresh PR.

I have beaten on this pretty hard... so I think that nothing is broken. But of course, you'll tell me if that is true or not.

So please take a look and give me your most awesomest feedback!

### CODE REVIEW

**Comment by gwenf** [comment]
Hey, thanks @camoore1979 
This seems to be working great. There is no working delete function for vehicles? Maybe that's coming later.

**Comment by BWortman** [comment]
Thanks, @camoore1979! It's a step in the right direction!

One bug I was able to replicate (usually):
1) Add a new vehicle, with or without the image.
2) The app will display the list of vehicles, including the new vehicle.
3) Click the VIN of the new vehicle. Crash:

TypeError: Cannot read property 'vin' of null
VehicleDetail.render
src/components/fleet/vehicles/VehicleDetail.js:16
  13 |     
  14 | return (
  15 |   <div>
> 16 |     <h3>{`Vehicle Detail: ${vehicle.vin}`}</h3>


Also, could you make Rent to Own a checkbox to match the spec?

File: src/components/common/form/dropZone.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        name={name}
+        accept="image/jpeg"
+        onDrop={(acceptedFiles) => {
+          encodeDropzoneImages(acceptedFiles)
+            .then((file) => input.onChange(file.imageSrc));    
+        }}
+        className={styles.dropzone}
+        multiple={false}
+      >
+        <div>Drag and drop image file here or click to select. (.jpeg only)</div>
```
Comment (BWortman) [inline src/components/common/form/dropZone.js]: Should use a button, to match the spec and the angular implementation.

File: src/components/common/utils/encodeDropzoneImages.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export default function encodeDropzoneImages(files) {
+  return new Promise((resolve, reject) => {
+    files.forEach((file) => {
+      const reader = new FileReader();
+      reader.onload = () => {
+        file.imageSrc = reader.result;
+        resolve(file);
+      };
+
+      reader.onabort = () => {
```
Comment (BWortman) [inline src/components/common/utils/encodeDropzoneImages.js]: Possibly silly question, but shouldn't this be onAbort? Similar question with onerror -> onError, below.

**Comment by camoore1979** [comment]
Thanks @gwenf and @BWortman !! Working the requested changes now.

**Comment by camoore1979** [comment]
Thanks for the reviews @BWortman @gwenf !!!

I have made the requested changes by @BWortman and also fixed #48 (I think they are now per spec).
Please give it a try!

**Comment by camoore1979** [comment]
@BWortman thank you for the thorough reviews & UAT! OK, I think this is it.... ;-)

Take a look and let's see if we can pull the trigger this time. Thank you!

**Comment by BWortman** [comment]
Just playing with tasks...
- [] - incomplete
- [x] - complete

ok, that's better

**Pull request merged**